internal/service/elbv2/listener.go

```diff
@@ -1937,6 +1933,71 @@ func listenerActionRuntimeValidate(actionPath cty.Path, action map[string]any, d
 	}
 }
 
+func validateMutualAuthenticationCustomDiff(_ context.Context, d *schema.ResourceDiff, _ any) error {
+	var diags diag.Diagnostics
+
+	configRaw := d.GetRawConfig()
+	if !configRaw.IsWhollyKnown() || configRaw.IsNull() {
+		return nil
+	}
+
+	mutualAuthPath := cty.GetAttrPath("mutual_authentication")
+	mutualAuth := configRaw.GetAttr("mutual_authentication")
+	if mutualAuth.IsKnown() && !mutualAuth.IsNull() {
+		mutualAuthListPlantimeValidate(mutualAuthPath, mutualAuth, &diags)
+	}
+
+	return sdkdiag.DiagnosticsError(diags)
+}
+
+func mutualAuthListPlantimeValidate(mutualAuthPath cty.Path, mutualAuth cty.Value, diags *diag.Diagnostics) {
+	it := mutualAuth.ElementIterator()
+	for it.Next() {
+		i, mutualAuth := it.Element()
+		mutualAuthPath := mutualAuthPath.Index(i)
+
+		mutualAuthPlantimeValidate(mutualAuthPath, mutualAuth, diags)
+	}
+}
+
+func mutualAuthPlantimeValidate(mutualAuthPath cty.Path, mutualAuth cty.Value, diags *diag.Diagnostics) {
+	modePath := mutualAuthPath.GetAttr(names.AttrMode)
+	mode := mutualAuth.GetAttr(names.AttrMode)
+	if !mode.IsKnown() {
+		return
+	}
+	if mode.IsNull() {
+		return
+	}
+
+	trustStoreARNPath := mutualAuthPath.GetAttr("trust_store_arn")
+	trustStoreARN := mutualAuth.GetAttr("trust_store_arn")
+
+	switch mode := mode.AsString(); mode {
+	case mutualAuthenticationVerify:
+		if trustStoreARN.IsNull() {
+			*diags = append(*diags, errs.NewAttributeRequiredWhenError(trustStoreARNPath, modePath, mode))
+		}
+
+	default:
+		advertisePath := mutualAuthPath.GetAttr("advertise_trust_store_ca_names")
+		advertise := mutualAuth.GetAttr("advertise_trust_store_ca_names")
+		if !advertise.IsNull() {
+			*diags = append(*diags, errs.NewAttributeConflictsWhenError(advertisePath, modePath, mode))
+		}
+
+		ignorePath := mutualAuthPath.GetAttr("ignore_client_certificate_expiry")
+		ignore := mutualAuth.GetAttr("ignore_client_certificate_expiry")
+		if !ignore.IsNull() {
+			*diags = append(*diags, errs.NewAttributeConflictsWhenError(ignorePath, modePath, mode))
+		}
+
+		if !trustStoreARN.IsNull() {
+			*diags = append(*diags, errs.NewAttributeConflictsWhenError(trustStoreARNPath, modePath, mode))
+		}
```

Comment:
Can we add acceptance tests which attempt to use these invalid argument combinations? Or alternatively unit tests for the parent `validateMutualAuthenticationCustomDiff`?